File: sync/src/main/scala/com/fan/mysql/event/impl/QueryEvent.java
package com.dingcloud.dts.binlog.event.impl;

import com.dingcloud.dts.binlog.event.AbstractEvent;
import com.dingcloud.dts.binlog.event.EventHeader;
import com.dingcloud.dts.binlog.parser.QueryEventParser.StatusVariable;

public class QueryEvent extends AbstractEvent {

	private long threadId;
	private long execTime;
	private int databaseLength;
	private int errorCode;
	private int statusVariablesLength;
	private StatusVariable statusVariables;
	private String databaseName;
	private String query;

	public QueryEvent(EventHeader header) {
		super(header);
	}

	public long getThreadId() {
		return threadId;
	}

	public void setThreadId(long threadId) {
		this.threadId = threadId;
	}

	public long getExecTime() {
		return execTime;
	}

	public void setExecTime(long execTime) {
		this.execTime = execTime;
	}

	public int getDatabaseLength() {
		return databaseLength;
	}

	public void setDatabaseLength(int databaseLength) {
		this.databaseLength = databaseLength;
	}

	public int getErrorCode() {
		return errorCode;
	}

	public void setErrorCode(int errorCode) {
		this.errorCode = errorCode;
	}

	public int getStatusVariablesLength() {
		return statusVariablesLength;
	}

	public void setStatusVariablesLength(int statusVariablesLength) {
		this.statusVariablesLength = statusVariablesLength;
	}

	public StatusVariable getStatusVariables() {
		return statusVariables;
	}

	public void setStatusVariables(StatusVariable statusVariables) {
		this.statusVariables = statusVariables;
	}

	public String getDatabaseName() {
		return databaseName;
	}

	public void setDatabaseName(String databaseName) {
		this.databaseName = databaseName;
	}

	public String getQuery() {
		return query;
	}

	public void setQuery(String query) {
		this.query = query;
	}

}

File: async/src/main/scala/com/fan/mysql/async/binlog/event/impl/QueryEvent.scala
package com.fan.mysql.async.binlog.event.impl

import com.fan.mysql.async.binlog.event.{AbstractEvent, EventHeader}
import com.fan.mysql.async.binlog.parse.QueryEventParser.StatusVariable

class QueryEvent(header: EventHeader) extends AbstractEvent(header) {
  private[this] var threadId = 0L
  private[this] var execTime = 0L
  private[this] var databaseLength = 0
  private[this] var errorCode = 0
  private[this] var statusVariablesLength = 0
  private[this] var statusVariables: StatusVariable = _
  private[this] var databaseName: String = _
  private[this] var query: String = _

  def getThreadId: Long = threadId

  def setThreadId(threadId: Long): Unit = {
    this.threadId = threadId
  }

  def getExecTime: Long = execTime

  def setExecTime(execTime: Long): Unit = {
    this.execTime = execTime
  }

  def getDatabaseLength: Int = databaseLength

  def setDatabaseLength(databaseLength: Int): Unit = {
    this.databaseLength = databaseLength
  }

  def getErrorCode: Int = errorCode

  def setErrorCode(errorCode: Int): Unit = {
    this.errorCode = errorCode
  }

  def getStatusVariablesLength: Int = statusVariablesLength

  def setStatusVariablesLength(statusVariablesLength: Int): Unit = {
    this.statusVariablesLength = statusVariablesLength
  }

  def getStatusVariables: StatusVariable = statusVariables

  def setStatusVariables(statusVariables: StatusVariable): Unit = {
    this.statusVariables = statusVariables
  }

  def getDatabaseName: String = databaseName

  def setDatabaseName(databaseName: String): Unit = {
    this.databaseName = databaseName
  }

  def getQuery: String = query

  def setQuery(query: String): Unit = {
    this.query = query
  }

}
